// @500apps/500apps-web/src/storage.js
class Storage {

    /**
     * Set session
     */
    setSessionStorage(key, value) {
        return sessionStorage.setItem(key, value);
    }

    /**
     * Get session
     */
    getSessionStorage(key) {
        return sessionStorage.getItem(key);
    }

    /**
     * Set session
     */
    setLocalStorage(key, value) {
        return localStorage.setItem(key, value);
    }

    /**
     * Get session
     */
    getLocalStorage(key) {
        return localStorage.getItem(key);
    }

    /**
     * Set Session Cookie
     *  
     * @param {String} cname - Cookie name
     * @param {String} cvalue - Cookie value to store
     * @param {String} exdays - Days set to expire
     */
    setCookie(cname, cvalue, exdays) {
        var d = new Date();
        d.setTime(d.getTime() + (exdays * 24 * 60 * 60 * 1000));
        let expires = ""

        // Expires is added only if exdays is set. Eg: session cookie does not have exdays. It's ephemeral.
        if (exdays && exdays > 0) {
            expires = "expires=" + d.toUTCString();
        }

        // Store in cookie
        document.cookie = cname + "=" + cvalue + ";" + expires + ";path=/";
    }

    /**
     * Generate Cookie
     */
    getCookie(cname) {
        var name = cname + "=";
        var decodedCookie = decodeURIComponent(document.cookie);
        var ca = decodedCookie.split(';');
        for (var i = 0; i < ca.length; i++) {
            var c = ca[i];
            while (c.charAt(0) == ' ') {
                c = c.substring(1);
            }
            if (c.indexOf(name) == 0) {
                return c.substring(name.length, c.length);
            }
        }
        return "";
    }

}
module.exports = Storage
